isis/src/qisis/objs/Footprint2DView/Footprint2DView.cpp

```diff
@@ -326,7 +317,25 @@ namespace Isis {
    * @param value The boolean that holds if a control network has been set.
    */
   void Footprint2DView::enableControlNetTool(bool value) {
-    m_controlNetTool->setEnabled(value);
+    foreach (QAction *action, m_toolPad->actions()) {
+      if (action->toolTip() == "Control Net (c)") {
+        action->setEnabled(value);
+      }
+    }
+  }
+
+
+  /**
+   * Enables toolbars and toolpad actions. Overriden method.
+   * If an active control network has not been set, do not enable the cnet tool.
+   */
+  void Footprint2DView::enableActions() {
+    foreach (QAction *action, actions()) {
+      if (action->toolTip() == "Control Net (c)" && !m_directory->project()->activeControl()) {
```

Comment:
`if (action->toolTip() == "Control Net (c)"`

Could this check be done with a qobject cast?

Reply:
I do not want to cast it into a MosaicControlNetTool because it does not inherit from QAction, so I cannot call setEnable(true) to enable the tool. I felt like this was the cleanest way. If you have more suggestions, let me know.